pkg/api/handlers/mcpgateway/handler.go

```diff
@@ -83,6 +83,10 @@ func (h *Handler) StreamableHTTP(req api.Context) error {
 		return fmt.Errorf("failed to get mcp server config: %w", err)
 	}
 
+	if mcpServer.Spec.NeedsURL {
+		return fmt.Errorf("mcp server %s needs to update its URL", mcpServer.Name)
+	}
```

Comment:
I like the idea of keeping this check in one area. You added it to `handlers.ServerForActionWithID`. If you also added it to `handlers.ServerFromMCPServerInstance`, then this check would be centralized to that file and we wouldn't need it here.

Reply:
As long as you are confident that that will suffice, then that sounds good to me.